From e781ec9049b962dcc670fa3b04868bc2d3944bef Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Wed, 19 Oct 2016 11:58:43 -0500
Subject: [PATCH 1229/5242] MLK-13361-1 arm: imx6q: busfreq: rename
 100MHz-related macros

commit  a8921da1075de8436747d28428d0d8fd24364d42 from
https://source.codeaurora.org/external/imx/linux-imx.git

Two macros are renamed:

1) set_timings_above_100MHz_operation as restore_mmdc_settings_info
2) mmdc_clk_lower_100MHz as mmdc_clk_lower_equal_100MHz

For (1) the operation is generic to several cases and not just related
(at least on a semantic way) with the operations "above" 100MHz

Renamed as restore_mmdc_settings_info the  macro can be reused for the
other cases like equal to 100MHz and possibly other intermediate
operation points.

For (2), the macro is renamed as mmdc_clk_lower_equal_100MHz to reflect
that this macro handles both the "lower than 100 MHz" case and the
"equal to 100MHz" case.

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr2_freq_imx6q.S |   21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-imx/lpddr2_freq_imx6q.S b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
index d032742..bff30c5c 100644
--- a/arch/arm/mach-imx/lpddr2_freq_imx6q.S
+++ b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
@@ -186,36 +186,35 @@ skip_below_100Mhz_ch1_timings:
 
 	.endm
 
-	.macro	set_timings_above_100MHz_operation
+	.macro restore_mmdc_settings_info
 	/* restore timing from mmdc_settings_info */
 	ldr	r6, [r1, #0x0]
 	ldr	r7, [r1, #0x4]
-tloop:
+1:
 	ldr	r9, [r7], #0x4
 	ldr	r10, [r7], #0x4
 	str	r10, [r8, r9]
 	subs	r6, r6, #0x1
-	bne	tloop
+	bne	1b
 
 	/* Check if lpddr2 channel 1 is enabled */
 	ldr	r6, [r8, #0x18]
 	ands	r6, r6, #(1 << 2)
-	beq	skip_above_100Mhz_ch1_timings
+	beq	3f
 
 	ldr	r6, [r1, #0x0]
 	ldr	r7, [r1, #0x4]
-tloop2:
+2:
 	ldr	r9, [r7], #0x4
 	ldr	r10, [r7], #0x4
 	str	r10, [r4, r9]
 	subs	r6, r6, #0x1
-	bne	tloop2
-
-skip_above_100Mhz_ch1_timings:
+	bne	2b
+3:
 
 	.endm
 
-	.macro	mmdc_clk_lower_100MHz
+	.macro	mmdc_clk_lower_equal_100MHz
 
 	set_timings_below_100MHz_operation
 	/*
@@ -305,7 +304,7 @@ skip_lower_force_measure_ch1:
 
 	.macro	mmdc_clk_above_100MHz
 
-	set_timings_above_100MHz_operation
+	restore_mmdc_settings_info
 
 	/* Make sure that the PHY measurement unit is NOT in bypass mode */
 	ldr	r5, =0x8B8
@@ -518,7 +517,7 @@ skip_sbs_ch1:
 	ldr	r10, =100000000
 	cmp	r0, r10
 	bgt	set_ddr_mu_above_100
-	mmdc_clk_lower_100MHz
+	mmdc_clk_lower_equal_100MHz
 
 set_ddr_mu_above_100:
 	ldr	r10, =24000000
-- 
1.7.9.5

